cmd/run.go

```diff
@@ -440,4 +551,313 @@ func executeRunWithAppsConfigFile(runFilePath string) {
+
+	// Wait for appRunnning channel output.
+	if appStarted := <-appRunning; !appStarted {
+		// Start App failed, try to stop Dapr and exit.
+		err = killDaprdProcess(runState)
+		return nil, err
+	}
+	return runState, nil
+}
+
+// stopDaprdAndAppProcesses is a function to stop the App process and the associated Daprd process
+// This should be called as a blocking function call.
+func stopDaprdAndAppProcesses(runState *runExec.RunExec) bool {
+	var err error
+	print.StatusEvent(runState.DaprCMD.OutputWriter, print.LogInfo, "\ntermination signal received: shutting down")
+	// Only if two different output writers are present run the following print statement.
+	if runState.AppCMD.OutputWriter != runState.DaprCMD.OutputWriter {
+		print.StatusEvent(runState.AppCMD.OutputWriter, print.LogInfo, "\ntermination signal received: shutting down")
+	}
+
+	exitWithError := false
+
+	daprErr := runState.DaprCMD.CommandErr
+
+	if daprErr != nil {
+		exitWithError = true
+		print.StatusEvent(runState.DaprCMD.ErrorWriter, print.LogFailure, "Error exiting Dapr: %s", daprErr)
+	} else if runState.DaprCMD.Command.ProcessState == nil || !runState.DaprCMD.Command.ProcessState.Exited() {
+		err = killDaprdProcess(runState)
+		if err != nil {
+			exitWithError = true
+		}
+	}
+	appErr := runState.AppCMD.CommandErr
+
+	if appErr != nil {
+		exitWithError = true
+		print.StatusEvent(runState.AppCMD.ErrorWriter, print.LogFailure, "Error exiting App: %s", appErr)
+	} else if runState.AppCMD.Command != nil && (runState.AppCMD.Command.ProcessState == nil || !runState.AppCMD.Command.ProcessState.Exited()) {
+		err = killAppProcess(runState)
+		if err != nil {
+			exitWithError = true
+		}
+	}
+	return exitWithError
+}
+
+// startAppsProcess, starts the App process and calls wait in a goroutine
+// This function should be called as a goroutine.
+func startAppProcess(runConfig *standalone.RunConfig, runE *runExec.RunExec,
+	appRunning chan bool, sigCh chan os.Signal, errorChan chan error,
+) {
+	if runE.AppCMD.Command == nil {
+		appRunning <- true
+		return
+	}
+
+	stdErrPipe, pipeErr := runE.AppCMD.Command.StderrPipe()
+	if pipeErr != nil {
+		print.StatusEvent(runE.AppCMD.ErrorWriter, print.LogFailure, "Error creating stderr for App %q : %s", runE.AppID, pipeErr.Error())
+		errorChan <- pipeErr
+		appRunning <- false
+		return
+	}
+
+	stdOutPipe, pipeErr := runE.AppCMD.Command.StdoutPipe()
+	if pipeErr != nil {
+		print.StatusEvent(runE.AppCMD.ErrorWriter, print.LogFailure, "Error creating stdout for App %q : %s", runE.AppID, pipeErr.Error())
+		errorChan <- pipeErr
+		appRunning <- false
+		return
+	}
+
+	errScanner := bufio.NewScanner(stdErrPipe)
+	outScanner := bufio.NewScanner(stdOutPipe)
+	go func() {
+		for errScanner.Scan() {
+			if runE.AppCMD.ErrorWriter == os.Stderr {
+				// Add color and prefix to log and output to Stderr.
+				fmt.Fprintln(runE.AppCMD.ErrorWriter, print.Blue(fmt.Sprintf("== APP == %s", errScanner.Text())))
```

Comment:
I think same comment as above at places where `fmt.Sprintf` can be replaced with `+`